js/main.js

```diff
@@ -28,25 +28,36 @@ const windowMgr = require('./windowMgr.js');
 app.on('ready', getUrlAndOpenMainWindow);
 
 function getUrlAndOpenMainWindow() {
-    getConfig().then(function(config) {
-        let protocol = '';
-        // add https protocol if none found.
-        let parsedUrl = nodeURL.parse(config.url);
-        if (!parsedUrl.protocol) {
-            protocol = 'https';
-        }
-        var url = nodeURL.format({
-            protocol: protocol,
-            slahes: true,
-            pathname: parsedUrl.href
-        });
-        windowMgr.createMainWindow(url);
-    }).catch(function(err) {
+    getConfig()
+    .then(createWin).catch(function (err){
         let title = 'Error loading configuration';
-        electron.dialog.showErrorBox(title, title + ': ' + err);
+        electron.dialog.showErrorBox(title, title + ': ' + err);            
     });
 }
 
+function createWin(podurl){
```

Comment:
the signature here should be config: createWin(config)

the idea is that getConfig will return object, for now it just return object with url but in future there could be other things here.